iree/base/BUILD

```diff
@@ -470,6 +470,7 @@ cc_library(
         "wait_handle_win32.c",
     ],
     hdrs = ["wait_handle.h"],
+    copts = ["-Wno-return-type"],
```

Comment:
We should fix this warning instead of disabling it. If we were to disable it, we should do so in a way that works across Bazel and CMake (at a minimum this would need to be reflected in iree/base/CMakeLists.txt as well).

cc @benvanik

Reply:
this is for an annoying warning.
it said

```
iree/base/wait_handle_poll.c:66:1: error: non-void function does not return a value in all control paths [-Werror,-Wreturn-type]
```
but I think all the cases are covered by the `if..else`

```
 if (rv > 0) {
    // One or more events set.
    *out_signaled_count = rv;
    return iree_ok_status();
  } else if (rv == 0) {
    // Timeout; no events set.
    return iree_status_from_code(IREE_STATUS_DEADLINE_EXCEEDED);
  } else if (IREE_UNLIKELY(rv < 0)) {
    return iree_make_status(iree_status_code_from_errno(errno),
                            "poll failure %d", errno);
  }
```

https://github.com/google/iree/blob/main/iree/base/wait_handle_poll.c#L55-L65

Reply:
The compiler doesn't know that one of the branches will be taken. We should just fix that rather than silence the warning.

Reply:
https://github.com/google/iree/pull/4041